src/Nox.Generator/Application/NoxWebApplicationExtensionGenerator.cs

```diff
@@ -13,69 +16,84 @@ public void Generate(SourceProductionContext context, NoxSolutionCodeGeneratorSt
-                    usings.Add("using Nox.EntityFramework.SqlServer;");
+                    namespaces.Add("Nox.EntityFramework.SqlServer");
                     dbProvider = "SqlServerDatabaseProvider";
                     break;
 
                 case DatabaseServerProvider.Postgres:
-                    usings.Add("using Nox.EntityFramework.Postgres;");
+                    namespaces.Add("Nox.EntityFramework.Postgres");
                     dbProvider = "PostgresDatabaseProvider";
                     break;
                 case DatabaseServerProvider.SqLite:
-                    usings.Add("using Nox.EntityFramework.Sqlite;");
+                    namespaces.Add("Nox.EntityFramework.Sqlite");
                     dbProvider = "SqliteDatabaseProvider";
                     break;
             }
         }
 
-        code.AppendLine("using Microsoft.EntityFrameworkCore;");               
-        code.AppendLine("using System.Reflection;");
-        code.AppendLine("using Microsoft.OData.ModelBuilder;");
-        code.AppendLine("using Nox;");
-        code.AppendLine("using Nox.Solution;");
-        code.AppendLines(usings.ToArray());
-        code.AppendLine("using Nox.Types.EntityFramework.Abstractions;");
-        code.AppendLine($"using {solution.Name}.Infrastructure.Persistence;");
-        if (config.Presentation)
-            code.AppendLine($"using {solution.Name}.Presentation.Api.OData;");
-        code.AppendLine();
-
-        code.AppendLine("public static class NoxWebApplicationBuilderExtension");
-        code.StartBlock();
-        code.AppendLine(@"public static IServiceCollection AddNox(this IServiceCollection services)
-                        {
-                            return services.AddNox(null);
-                        }
-                        ");
-        code.AppendLine("public static IServiceCollection AddNox(this IServiceCollection services, Action<ODataModelBuilder>? configureOData)");
-        code.StartBlock();
-        code.AppendLine($"services.AddNoxLib(Assembly.GetExecutingAssembly());");
-        code.AppendLine("services.AddNoxOdata(configureOData);");
-        var dbContextName = $"{solution.Name}DbContext";
+        namespaces.Sort();
 
-        code.AppendLine($"services.AddSingleton(typeof(INoxClientAssemblyProvider), s => new NoxClientAssemblyProvider(Assembly.GetExecutingAssembly()));");
-        code.AppendLine($"services.AddSingleton<DbContextOptions<{dbContextName}>>();");
-        code.AppendLine($"services.AddSingleton<INoxDatabaseConfigurator, {dbProvider}>();");
-        code.AppendLine($"services.AddSingleton<INoxDatabaseProvider, {dbProvider}>();");
-        code.AppendLine($"services.AddDbContext<{dbContextName}>();");
-        code.AppendLine($"services.AddDbContext<DtoDbContext>();");
-        code.AppendLine("return services;");
-        code.EndBlock();
-        code.AppendLine();
-
-        code.EndBlock();
+        object model = new
```

Comment:
I could add more variables to the template and send them in the model. I could, for example, send all the `services.Add` and make it a loop like I did in the Namespaces. But I'm don't know how much generic this template should be since we are only using it here.